pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -35,14 +36,24 @@ const (
 	GroupType = "group"
 )
 
+type tokenManager interface {
+	IsMemberOf(token v3.Token, group v3.Principal) bool
```

Comment:
I'm fine with this, but you don't need the following methods:

 * GetGroupsForTokenAuthProvider
 * UpdateToken
 * GetSecret

As they're not called by the OpenIDCProvider?

Reply:
Good point! We are using `GetSecret` in the keycloak oidc provider, but I removed `GetGroupsForTokenAuthProvider` and `UpdateToken`

Reply:
Right, but the `OpenIDCProvider` doesn't need `GetSecret` and this interface is only used for that?

Reply:
keycloak oidc provider embeds the `OpenIDCProvier`, and it uses  `GetSecret` [here](https://github.com/raulcabello/rancher/blob/keycloak-fix/pkg/auth/providers/keycloakoidc/keycloak_provider.go#L160)

Reply:
Doh! sorry, yes, didn't spot that :-)